psi4/driver/procrouting/dft_funcs/superfuncs.py

```diff
@@ -182,6 +187,23 @@ def build_superfunctional(name, restricted):
     # Lock and unlock the functional
     sup[0].set_lock(False)
 
+    # pre-defined -NL dashparam first, then special dft_disp_parm option.
+    # not applicable for XC generating function
+    if not hasattr(name, '__call__'):
+        if (any(name.lower().endswith(al) for al in dftd3.full_dash_keys)):
+            dashparam = [x for x in dftd3.full_dash_keys if name.endswith(x)]
+            if (dashparam[0] == 'nl'):
+                base_name = name.replace('-' + dashparam[0], '')
+                sup[0].set_vv10_b(dftd3.dashcoeff['nl'][base_name.lower()]['b'])
+                sup[0].set_vv10_c(dftd3.dashcoeff['nl'][base_name.lower()]['c'])
+                if (core.has_option_changed("SCF", "DFT_DISPERSION_PARAMETERS")):
```

Comment:
As above; they could be passed into the `tuple_params` argument to empirical dispersion class, and then processed inside.